lib/codegen/ProofEvent.cpp

```diff
@@ -372,13 +392,42 @@ std::pair<llvm::BasicBlock *, llvm::BasicBlock *> proof_event::proof_branch(
   return {true_block, merge_block};
 }
 
+std::pair<llvm::BasicBlock *, llvm::BasicBlock *> proof_event::proof_branch(
```

Comment:
this feels pretty copy-pasted; is there a way we could separate out a more generic function? maybe using templates?